openghg/standardise/boundary_conditions/_openghg.py

```diff
@@ -0,0 +1,96 @@
+    """
+    Parses the boundary conditions file and adds data and metadata.
+    Args:
+        filepath: Path of boundary conditions file
+        species: Species name
+        bc_input: Input used to create boundary conditions. For example:
+            - a model name such as "MOZART" or "CAMS"
+            - a description such as "UniformAGAGE" (uniform values based on AGAGE average)
+        domain: Region for boundary conditions
+        chunks: Chunking schema to use when storing data. It expects a dictionary of dimension name and chunk size,
+                for example {"time": 100}. If None then a chunking schema will be set automatically by OpenGHG.
+                See documentation for guidance on chunking: https://docs.openghg.org/tutorials/local/Adding_data/Adding_ancillary_data.html#chunking.
+                To disable chunking pass in an empty dictionary.
+    Returns:
+        Dict: Dictionary of "species_bc_input_domain" : data, metadata, attributes
+    """
+    species = clean_string(species)
+    species = synonyms(species)
+    bc_input = clean_string(bc_input)
+    domain = clean_string(domain)
+
+    filepath = Path(filepath)
+
+    with xr.open_dataset(filepath).chunk(chunks) as bc_data:
+        # Some attributes are numpy types we can't serialise to JSON so convert them
+        # to their native types here
+        attrs = {}
+        for key, value in bc_data.attrs.items():
+            try:
+                attrs[key] = value.item()
+            except AttributeError:
+                attrs[key] = value
+
+        author_name = "OpenGHG Cloud"
+        bc_data.attrs["author"] = author_name
+
+        metadata = {}
+        metadata.update(attrs)
+
+        metadata["species"] = species
+        metadata["domain"] = domain
+        metadata["bc_input"] = bc_input
+        metadata["author"] = author_name
+        metadata["processed"] = str(timestamp_now())
+
+        # Check if time has 0-dimensions and, if so, expand this so time is 1D
+        if "time" in bc_data.coords:
+            bc_data = update_zero_dim(bc_data, dim="time")
+
+        bc_time = bc_data["time"]
+
+        start_date, end_date, period_str = infer_date_range(
+            bc_time, filepath=filepath, period=period, continuous=continuous
+        )
+
+        metadata["start_date"] = str(start_date)
+        metadata["end_date"] = str(end_date)
+
+        metadata["max_longitude"] = round(float(bc_data["lon"].max()), 5)
+        metadata["min_longitude"] = round(float(bc_data["lon"].min()), 5)
+        metadata["max_latitude"] = round(float(bc_data["lat"].max()), 5)
+        metadata["min_latitude"] = round(float(bc_data["lat"].min()), 5)
+        metadata["min_height"] = round(float(bc_data["height"].min()), 5)
+        metadata["max_height"] = round(float(bc_data["height"].max()), 5)
+        metadata["input_filename"] = filepath.name
+
+        metadata["time_period"] = period_str
+
+        key = "_".join((species, bc_input, domain))
```

Comment:
This line is the only reason species, bc_input, and domain are needed by this parser. Maybe they should be removed from `metadata` so that when the outputs of `standardise_` are reworked, they will show up as unused arguments.

Reply:
It's a good idea but doesn't quite work at the moment because the `additional_input_parameters` are filtered based on what the parse function accepts. Because it accepts `domain`, `bc_input`, it then doesn't appear in the `additional_input_parameters` and so isn't added afterwards. Once `domain`, `bc_input` etc. are removed from the `parse_openghg` signature this would work though.

Reply:
Ah right, okay